### DISCUSSION TITLE
podman-OpenFile demo: using `OpenFile=/home/test/sockdir/sock` to give a container process access to a unix socket

### DISCUSSION THREAD

**Discussion by eriksjolund** [containers/podman#18126] [Show and tell]
The systemd directive [`OpenFile=`](https://www.freedesktop.org/software/systemd/man/systemd.service.html#OpenFile=) was introduced in __systemd 253__ (released 15 February 2023).

I created a demo 

https://github.com/eriksjolund/podman-OpenFile

that deals with this situation:

__Problem__: A container process does not have file permissions to access the UNIX socket that a web server listens on.

__Solution__: Start the container with

```
systemd-run
  --quiet \
  --property OpenFile=$HOME/sockdir/sock:fdnametest \
  --user \
  --collect \
  --pipe \
  --wait \
  podman \
    run \
    --rm \
    --user 65534:65534 \
    localhost/demo /demo http://localhost
```

so that __systemd__ connects to the UNIX socket. The container process inherits the established socket.

__Previous discussion:__ https://github.com/containers/podman/discussions/17789

**Comment by rhatdan** [comment]
You need to translate this into a Blog.

I would think SELinux would block this access, BTW.

**Reply by eriksjolund** [reply]
Good idea, I should translate it to a blog.
